packages/flagging/src/openfeature/provider.ts

```diff
@@ -5,57 +5,68 @@
 import type { EvaluationContext, JsonValue, Logger, Paradigm, ResolutionDetails } from '@openfeature/core'
 import { StandardResolutionReasons } from '@openfeature/core'
 import type { Provider } from '@openfeature/web-sdk'
+import type { PrecomputeClient } from '../precomputeClient'
 
 export class DatadogProvider implements Provider {
   runsOn: Paradigm = 'client'
   metadata = {
     name: 'datadog',
   } as const
 
+  private precomputeClient: PrecomputeClient | undefined
+
+  constructor(precomputeClient?: PrecomputeClient) {
+    this.precomputeClient = precomputeClient
+  }
+
   resolveBooleanEvaluation(
-    _flagKey: string,
+    flagKey: string,
     defaultValue: boolean,
     _context: EvaluationContext,
     _logger: Logger
   ): ResolutionDetails<boolean> {
     return {
-      value: defaultValue,
+      value: this.precomputeClient?.getBooleanAssignment(flagKey, defaultValue) ?? defaultValue,
       reason: StandardResolutionReasons.DEFAULT,
     }
   }
 
   resolveStringEvaluation(
-    _flagKey: string,
+    flagKey: string,
     defaultValue: string,
     _context: EvaluationContext,
     _logger: Logger
   ): ResolutionDetails<string> {
     return {
-      value: defaultValue,
+      value: this.precomputeClient?.getStringAssignment(flagKey, defaultValue) ?? defaultValue,
       reason: StandardResolutionReasons.DEFAULT,
     }
   }
 
   resolveNumberEvaluation(
```

Comment:
OF doesn't differentiate between integer and floating point types. We will need to handle this in a future interation.

Reply:
Interesting. We've seen some requests for write-time validation. Perhaps we ditch integer flags but offer integers as one of those validations. Or push to expand the OF spec 🤔

Reply:
Yea .. I am thinking the former. Although types like `double` exist I don't see a ton of reasons why customers would care to create a flag with one type and not the other, especially since we don't allow type changes. I bet its pretty frustrating to have a value like `5`, want another variation `5.5` and be blocked from creating it